runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java

```diff
@@ -259,28 +278,44 @@ static StreamExecutionEnvironment createStreamExecutionEnvironment(
         options.setShutdownSourcesAfterIdleMs(0L);
       }
     }
+  }
 
-    applyLatencyTrackingInterval(flinkStreamEnv.getConfig(), options);
-
-    if (options.getAutoWatermarkInterval() != null) {
-      flinkStreamEnv.getConfig().setAutoWatermarkInterval(options.getAutoWatermarkInterval());
-    }
-
-    // State backend
-    if (options.getStateBackendFactory() != null) {
+  private static void configureStateBackend(
+      FlinkPipelineOptions options, StreamExecutionEnvironment env) {
+    if (options.getStateBackend() != null) {
+      final String storagePath = options.getStateBackendStoragePath();
+      Preconditions.checkArgument(
+          storagePath != null,
+          "State backend was set to '%s' but no storage path was provided.",
+          options.getStateBackend());
+
+      final StateBackend stateBackend;
+      if (options.getStateBackend().equalsIgnoreCase("rocksdb")) {
```

Comment:
Should not this be the classname of the backend? I am assuming that what we want to achieve here is to allow new backends (not available in Flink) to be configured too (see comment above)

Reply:
There is no factory method available for state backends, they all have different constructors. We can't use the class name.

Reply:
I've reduced this to one call by adding a null check.